Polish Index

tajmone opened the pull request:
Edit "Overview": 
- Rephrased, shorter and possibly slightly smoother.
- A few typos fixed.

Edit "Features": 
- Some polishing and typos corrections.
- "un-highlight" -> "unhighlight" — reference:
   https://english.stackexchange.com/questions/192462/what-is-the-meaning-of-unhighlight

tajmone commented:
Hi Isaac! After you pointed out to me (on SublimeText forum) this great package I took interest and started to look into it.

Since I'm reading the docs for the first time (a "clean slate" approach) I would like to contribute by editing those parts of the documentation which required me to read them over again — ie: sentences that could be disentangled a bit and made smoother on the reader, plus the occasional typo here and there.

If it's ok with you, as I carry on reading the docs I'll also edit the markdown sources and add more commits.

facelessuser commented:
> Since I'm reading the docs for the first time (a "clean slate" approach) I would like to contribute by editing those parts of the documentation which required me to read them over again — ie: sentences that could be disentangled a bit and made smoother on the reader, plus the occasional typo here and there.

Please do.  I'll be the first to admit I hate writing documents, and the quality can lack in some places.

I may need you to rebase at some point as the new version of mkdocs seems to have broken my spellcheck and such.  I'll try and resolve the general Travis issues to get a sound base you can continue off of.

facelessuser commented:
Master no longer fails document tests, so if you could rebase to the tip of master, that'd be great.  Had to limit the version of the document theme and update the document config as it was breaking the build.

tajmone commented:
> I may need you to rebase at some point as the new version of mkdocs seems to have broken my spellcheck and such. I'll try and resolve the general Travis issues to get a sound base you can continue off of.

No problems. I'm just commiting as I go along reading because I prefer to work on first impressions — I work _a lot_ with documentation (I love it!) and I know from first hand experience how personal envolvement with the text creates blind areas: once you've digested the contents, it's like having translated them into a mental picture, and when you reread them you are influenced by your previous understanding and tend to navigate the text by concepts, rather then the actual phrases constructs. So a "clean slate" first reading is always advantageous.

facelessuser commented:
100%.  The brain often replaces what I meant than sometimes what I actually wrote.  Take your time, and I really appreciate your time to improve the docs.

facelessuser commented:
In general, I am not a fan of using `+` for a list.  I prefer `-`, so I'd rather not have things like that changed.

facelessuser commented on docs/src/markdown/contributing.md:
```diff
@@ -7,11 +7,11 @@ Sublime Versions | Description
-- Code patches via pull requests.
+- Bug reports,
+- Code reviewing,
+- Code patches via pull requests,
```
Is this is a list, I find the commas odd.  Leaving off punctuation would have been less odd

tajmone replied:
Sorry for "+" change, it was auto-introduced by the editor.

facelessuser commented on docs/src/markdown/contributing.md:
```diff
@@ -57,7 +57,7 @@ There are a couple of dependencies that must be present before running the tests
 1. As ST3 is the only current, actively supported version, Python 3.3 must be used to validate the tests.
 
-2. Unit tests are run with `pytest`.  You can install pytest via:
+2. Unit tests are run with `pytest`.  You can install [pytest](https://pytest.org) via:
```
I actually have a snippet that contains links: https://github.com/facelessuser/RegReplace/blob/master/docs/src/markdown/_snippets/links.md.  I prefer links like this to be placed there and then just do something like `[pytest][pytest]`.  That way we can reuse the link as an easy to remember key if desired.

tajmone replied:
I noticed you used reference style links in other places, but I couldn't find their definitions at the end of the doc, so I just pasted the links in place. I'll keep in mind this for the future.

facelessuser commented on docs/src/markdown/contributing.md:
```diff
@@ -85,7 +85,7 @@ There are a couple of dependencies that must be present before running the tests
 ## Documentation Improvements
 
-A ton of time has been spent not only creating and supporting this plugin, but also spent making this documentation.  If you feel it is still lacking, show your appreciation for the plugin by helping to improve the documentation.  Help with documentation is always appreciated and can be done via pull requests.  There shouldn't be any need to run validation tests if only updating documentation.
+A huge amount of time has been spent not only creating and supporting this plugin, but also redacting this documentation.  If you feel it's still lacking, show your appreciation for the plugin by helping to improve the documentation.  Help with documentation is always appreciated and can be done via pull requests.  There shouldn't be any need to run validation tests when updating just the documen ...
```
I'd prefer in some sense in keep the vocabulary a bit more accessible to the common person, so something other than `redacting` would be better.

tajmone replied:
What about "writing the documentation"? The verbs "do" and "make" have a really broad in meaning. 

> PS: Don't get me wrong on such corrections, they are not about showing off vocabulary or being too meticulus, it has to do with the fact that I've studied a bit the psychological aspects of comunication (especially reading) and as a general rule when the reader finds broad terms or undefined statements, in the back of his mind he keeps wondering with questions (which distract him from reading on).
> 
> As an example, if I tell you that I paid "something a lot of money", you'll be left wondering how much "a lot" really is. Similar unanswered question tend to build up and distract the flow of reading, and they also apply to smaller details. Of course, in narrative this might be something desired (where you want to reader to freely fantasize in the interpretation of the text and loose his sense of time), but in technical documentation you're looking for a smooth and linear path with as little bumps on the road as possible, because the reader's goal is about learning and memorizing, and such texts tend to be more cumbersome than other readings.

facelessuser commented on docs/src/markdown/installation.md:
```diff
@@ -6,26 +6,26 @@ The recommended way to install RegReplace is via [Package Control][package-contr
 1. Ensure Package Control is installed.  Instructions are found [here][package-control-install].
 
-2. In Sublime Text, press <kbd>ctrl</kbd> + <kbd>shift</kbd> + <kbd>p</kbd> (Win, Linux) or <kbd>cmd</kbd> + <kbd>shift</kbd> + <kbd>p</kbd> (OSX) to bring up the quick panel and start typing `Package Control: Install Package`.  Select the command and it will show a list of installable plugins.
+2. In Sublime Text, press <kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>p</kbd> (Win, Linux) or <kbd>CMD</kbd> + <kbd>Shift</kbd> + <kbd>p</kbd> (OSX) to bring up the quick panel and start typing `Package Control: Install Package`.  Select the command and it will show a list of installable plugins.
```
I'd omit this as I have a plugin that should convert this and add symbols to the key.  So I'll update this to be something like `++ctrl+shift+p++` and let the plugin handle this.

facelessuser commented on docs/src/markdown/usage.md:
````diff
@@ -368,16 +368,16 @@ When `selection_only` is enabled, you might have a regular expression chain that
 ```
 
-## Apply Regular Expressions Right Before File Save Event
+## Apply Regular Expressions Right before File Save Event
````
Before should probably remain title cased here.

tajmone replied:
Depends which style sytem you follow. I passed the titles through the Chicago style (CMS), which converts "before" to lowercase — MLA style too. The APA and AP styles would have it as "Before":

https://capitalizemytitle.com/

As a long as consistent styling is applied, anyone of them should be ok. I went for Chicago style because it's not too academic.

facelessuser commented on docs/src/markdown/usage.md:
```diff
@@ -575,7 +571,7 @@ You can read more about the backrefs' features in the [Backrefs' documentation][
 It is not always clear when Package Control updates dependencies.  So to force dependency updates, you can run Package Control's `Satisfy Dependencies` command which will update to the latest release.
 
-### Using Backrefs in RegReplace Plugin
+### Using Backrefs in Regreplace Plugins
```
RegReplace is the correct casing of the plugin name.

tajmone replied:
Sorry for that, the title capitalization tools should have preserved manually introduced capitalizations, but apparently it didn't. *Mea culpa* for not checking.

facelessuser commented:
I realize you are still moving through this, but I figured I'd go ahead and leave some comments.

tajmone commented:
> I realize you are still moving through this, but I figured I'd go ahead and leave some comments.

Please, do! this will help mutual understanding and render future collaboration easier — once I get know your style preferences, it becomes easier for me to make editing choices.

I have read all the text, so I think we can fix the pendig corrections and merge.

How we should we go about it? Shall I integrate in a new commit all the corrections you pointed out, and then you merge; or do you need me to squash all this into a single commit?

tajmone commented:
With commit 5e5cebd I've applied all your suggested fixes (and corrections rollbacks).

I can't preview the rendered reference styled links, but I've fixed them and added their definitions in `links.md` and they should fine.

facelessuser commented:
You can preview the whole site if you use mkdocs, but you'll have to rebase to fix document issues with the latest release.

Anyways, whenever you're done, just rebase on master, and then I will finish reviewing this.

tajmone commented:
> You can preview the whole site if you use mkdocs, but you'll have to rebase to fix document issues with the latest release.

I use other toolchains to build markdown documentation.

> Anyways, whenever you're done, just rebase on master, and then I will finish reviewing this.

Done.

facelessuser commented:
Travis CI is showing a spelling issue. If the word is okay, it can be added to `.dictonary`, if not, just correct it.

facelessuser commented:
> I use other toolchains to build markdown documentation.

In your case, I don't think you *need* need to preview this, but using different toolchains doesn't matter as mkdocs can be installed along side whatever, and if you need to keep your environment unchanged, you can always use `virtualenv` to set up a simple mkdocs environment, and delete it when done.  But like I said, I don't think that is necessary here as nothing structural has been changed.

facelessuser commented on docs/src/markdown/contributing.md:
```diff
@@ -37,16 +37,16 @@ Take part in reviewing pull requests and/or reviewing direct commits.  Make sugg
+Pull requests are welcome, and if you plan on contributing directly to the code, there are a couple of things you should bare in mind.
 
-Continuous integration tests on are run on all pull requests and commits via Travis CI.  When making a pull request, the tests will automatically be run, and the request must pass to be accepted.  You can (and should) run these tests before pull requesting.  If it is not possible to run these tests locally, they will be run when the pull request is made, but it is strongly suggested that requeste ...
+Continuous integration tests on are run on all pull requests and commits via [Travis CI][travis].  When making a pull request, the tests will be automatically run, and the request must pass to be accepted.  You can (and should) run these tests locally before pull requesting.  If it's not possible to run them locally, they will be run when the pull request is made, but it's strongly recommend that ...
```
Just something I noticed (not that you introduced):

Continuous integration tests ~~on~~ are run

tajmone replied:
Also:

"but it's strongly recommend that" > "recommended"

facelessuser commented on docs/src/markdown/installation.md:
```diff
@@ -10,22 +10,22 @@ The recommended way to install RegReplace is via [Package Control][package-contr
     If you are forking for a pull request, you should **just** clone RegReplace and run Package Control's `Satisfy Dependency` command to get all the dependencies.
 
-For those who want to install RegReplace without package control, here are the steps.  It understood that some people for what ever reason will prefer manually and may even have legitimate reasons to do so.  When going this route, you will have to keep all the packages updated yourself.
+For those who want to install RegReplace without package control, here are the steps.  Some people prefer manual installation — and may have legitimate reasons to do so.  When going this route, you will have to keep all the packages updated yourself.
```
General note: I usually prefer using `--` which smarty will convert to the Unicode dash.  I often can't tell which is being is used `—` or `-` when looking at documents with certain fonts.

facelessuser replied:
To be honest, I'm not sure an emdash makes sense here being followed by `and`.  I think the emdash can be removed altogether.

facelessuser replied:
I actually think we can remove this sentence: `Some people prefer manual installation — and may have legitimate reasons to do so.`

tajmone replied:
I agree with removing the sentence, it doesn't really contribute much — possibly, users choosing manual installation are already expert enough to know what they are doing and why; unexperienced users, on the other hand, probably won't care about the reasons why others do what's beyond their understanding.

As for the dashes, I'm just so used to manually enter them as chars that I do it automatically. I think it's better using the actual char because some MD converters/previews don't deal with smart punctuation (for example here, in GH issues, the "--" and "---" aren't converted. Technically, markdown docs should contain stricly ASCII chars only, but in practice everyone uses UTF8 freely in markdown docs, and very few parsers will object to it.

facelessuser replied:
> agree with removing the sentence, it doesn't really contribute much — possibly, users choosing manual installation are already expert enough to know what they are doing and why; unexperienced users, on the other hand, probably won't care about the reasons why others do what's beyond their understanding.

I sometimes write what I'm thinking even if it doesn't contribute much.  It's only times like this when I'm reviewing it that I realize it was kind of a pointless statement.

> I think it's better using the actual char because some MD converters/previews don't deal with smart punctuation (for example here, in GH issues, the "--" and "---" aren't converted. 

I'll be honest, I'm not concerned with what other parsers would do with *these* docs.  I try to explain up front that I expect them to be rendered in the mkdocs environment with the specified toolchain.  I do have a biased self interest in the way I format my docs.  I even use custom extensions to assist in doing custom things. I'm not against Unicode in Markdown, but as stated in my documents, emdash and `-` are difficult to distinguish with certain fonts.  The toolchain is easy to install, `pip install mkdocs-material` will actually install all the requirements. And with `virtualenv` you can even leave your default Python environment untouched.

Even GitHub does its own thing as GFM is *not* Markdown, but a Markdown-ish derivative.  Though they may be going down a more CommonMark path now, but even that is Markdown-ish as well, even if it is trying to establish itself as the One True Markdown.

facelessuser commented on docs/src/markdown/index.md:
```diff
@@ -12,7 +12,7 @@ RegReplace is a plugin for Sublime Text that allows the creating of commands con
-- Create find and mark/unmark commands to simply highlight or un-highlight results instead of replacing them.
+- Create commands that highlight results and require confirmation before replacing.
+- Create find and fold/unfold commands to fold or unfold the found results instead of replacing them.
+- Create find and mark/unmark commands to simply highlight or unhighlight results instead of replacing them.
```
Just add `unhighlight` to `.dictionary`.

tajmone replied:
Done.

facelessuser commented:
Changes look good! Thanks again for your work on this.  It is much appreciated.

tajmone commented:
You're welcome! I enjoyed it and I'm really grateful for this great package ... so, whichever way I contribute to it I'm happy to.
